qiskit/primitives/backend_sampler_v2.py

```diff
@@ -213,7 +219,9 @@ def _postprocess_pub(
         meas = {
             item.creg_name: BitArray(arrays[item.creg_name], item.num_bits) for item in meas_info
         }
-        return SamplerPubResult(DataBin(**meas, shape=shape), metadata={})
+        return SamplerPubResult(
+            DataBin(**meas, shape=shape), metadata={"circuit_metadata": circuit_metadata}
```

Comment:
Shouldn't this have `shots` too? Given `EstimatorPubResult` has `target_precision`. We should update the QRT Sampler to also return shots.

Reply:
Thank you for reviewing this PR.

Qiskit Runtime Sampler V2 does not have `shots` in the metadata because it can be obtained by `result[i].data.{creg}.num_shots`. I removed `shots` from the metadata of `StatevectorSampler` in this PR because of the same reason.
On the other hand, Estimator V2 has `shots` in the metadata.

SamplerV2 of qiskit-ibm-runtime returns the following metadata with the default option.
```
result.metadata = {'version': 2}
result[i].metadata = {'circuit_metadata': {}}
```

EstimatorV2 of qiskit-ibm-runtime returns the following metadata

```
result.metadata = {'dynamical_decoupling': {'enable': False, 'sequence_type': 'XX', 'extra_slack_distribution': 'middle', 'scheduling_method': 'alap'}, 'twirling': {'enable_gates': False, 'enable_measure': True, 'num_randomizations': 'auto', 'shots_per_randomization': 'auto', 'interleave_randomizations': True, 'strategy': 'active-accum'}, 'resilience': {'measure_mitigation': True, 'zne_mitigation': False, 'pec_mitigation': False}, 'version': 2}
result[i].metadata = {'shots': 128, 'target_precision': 0.1, 'circuit_metadata': {}, 'resilience': {}, 'num_randomizations': 2}
```